packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -624,7 +624,11 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
 
           onUpdateCallbacks.forEach(({ range, attrs, text }) => {
             const { doc, selection } = tr;
-            this.options.onUpdateLink(text, { doc, selection, range, attrs });
+
+            // Push to the end of the execution queue to prevent possible DOMException
+            setTimeout(() => {
```

Comment:
Did you try any alternatives here? Such as `requestAnimationFrame`?

Reply:
I have not as I mentioned in the [issue](https://github.com/remirror/remirror/issues/1754#issue-1287020628) I am not even sure if this is a bug or intentional. 

Don't think `requestAnimationFrame` would work though, have you looked at the code sandbox?

Reply:
> Don't think `requestAnimationFrame` would work though

Looks like it does, you're correct that this is being fired at the wrong point in time though. It means no commands can be trigger from in the event handler, as it is executed against stale state.

P.s. I would rather not delay in a loop, I  would recommend wrapping the whole block instead recommend this was changed to

```ts
window.requestAnimationFrame(() => {
  onUpdateCallbacks.forEach(({range, attrs, text}) => {
    const {doc, selection} = tr;
    this.options.onUpdateLink(text, {doc, selection, range, attrs});
  });
});
```

Reply:
>  I would recommend you change your code to use the dedicated hooks, rather than using the manager directly.


Thank you for the recommendation.


> Looks like it does, you're correct that this is being fired at the wrong point in time though. It means no commands can be executed within the event handler, as it is executed against stale state.

Great let me push a commit